### PR TITLE
linux: add OptionalsReg for audio drivers

### PR DESCRIPTION
**Pull request by FernetMenta**
At the time we added Register for audio drivers we had no folder platform/linux

### CODE REVIEW

**Comment by FernetMenta** [comment]
@pkerling @lrusak you may want to adapt Wayland and gbm

**Comment by Rechi** [comment]
`xbmc/platform/linux/OptionalsReg.(cpp|h)` files are missing.

File: xbmc/platform/linux/OptionalsReg.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+//-----------------------------------------------------------------------------
+// PulseAudio
+//-----------------------------------------------------------------------------
+
+#ifdef HAS_PULSEAUDIO
+#include "cores/AudioEngine/Sinks/AESinkPULSE.h"
+bool OPTIONALS::PulseAudioRegister()
+{
+  bool ret = CAESinkPULSE::Register();
+  return ret;
```
Comment (pkerling) [inline xbmc/platform/linux/OptionalsReg.cpp]: What is the temporary variable for?

Comment (FernetMenta) [reply xbmc/platform/linux/OptionalsReg.cpp]: @fritsch preferred this style. I don't think it's bad either.

Comment (lrusak) [reply xbmc/platform/linux/OptionalsReg.cpp]: https://stackoverflow.com/questions/3649468/how-to-set-a-breakpoint-in-gdb-where-the-function-returns

Comment (fritsch) [reply xbmc/platform/linux/OptionalsReg.cpp]: Known - I prefer it lazy - thx :-)

**Pull request merged**